### PR TITLE
Improve README with ftdetect gotcha

### PR DESCRIPTION
**Pull request by ixti**
### See:

- https://github.com/slim-template/vim-slim/issues/38
- https://github.com/slim-template/vim-slim/commit/1ac9ebd51467ddbe0f12f01cdac57a76483a00af

### CODE REVIEW

**Comment by rapind** [comment]
Interesting to note that we had `set filetype=slim` in master a couple years ago, but it was changed back for some reason (not sure why). See: https://github.com/slim-template/vim-slim/commit/1ac9ebd51467ddbe0f12f01cdac57a76483a00af

**Comment by ixti** [comment]
@igas @rapind What do you think about the PR now?

**Comment by ixti** [comment]
Instead of brute forcing filetype, I think we should just inform users of the issue and how to fix it.

**Comment by rapind** [comment]
@ixti I definitely think this is an improvement. It's kind of a strange use case that has you scratching your head, so documenting it is definitely nice for user's that hit it.

File: README.md [hunk 1]
```diff
@@ -62,3 +62,29 @@ If you're adding Vundle to a built-up vimrc, just make sure all these calls
 3. Open vim and run `:PluginInstall`.
 
 To update, open vim and run `:PluginInstall!` (notice the bang!)
+
+
+Known Issues
+------------
+
+We use `setfiletype` upon autodetect, which does not overrides filetype once it
```
Comment (rapind) [inline README.md]: did you mean `which overrides previous values` instead of `which does not overrides`?

Comment (rapind) [inline README.md]: `set fieltype` does override. `setf` does not.

File: ftdetect/slim.vim [hunk 2]
```diff
@@ -1 +1 @@
-autocmd BufNewFile,BufRead *.slim setf slim
+autocmd BufNewFile,BufRead *.slim setfiletype slim
```
Comment (rapind) [inline ftdetect/slim.vim]: Missing a space and an =. Should be `set filetype=slim`

Comment (ixti) [inline ftdetect/slim.vim]: I can't tell if you are trolling or not :D
The idea is to stick with `setf[ilename]` by default, as it (!) does not overrides filetype if it was set. If we will switch to `set filetype=slim` then user won't be able to "change" our default behavior easily.

Comment (rapind) [inline ftdetect/slim.vim]: In that case `setf` was fine. I was just confused as to why you changed it. Not trolling :)

**Comment by ixti** [comment]
@rapind although I understand your point of view (and I'm actually gonna use `set filetype=slim` in my `.vimrc`). I think there is a pretty small but still possible chance of causing issue for some potential user of this plugin if we will go obtrusive way.

And personally I think that any plugin should do as little as possible by default. So that user would configure/enable necessary behavior himself as he needs.

**Comment by rapind** [comment]
I've also changed it locally. I just think it'll impact more people by being unobtrusive, and I haven't been able to think of a scenario where being obtrusive would impact anyone (why would they assign something else to .slim?).

I'm happy enough with a mention in the readme though. We've probably discussed it to death at this point :)

File: ftdetect/slim.vim (hunk 2 above)
Comment (ixti) [inline ftdetect/slim.vim]: Ah.. I simply prefer longer versions (when it's not making too much noise), as it helps to get overall understanding of what' going on without calling `:help {thing}` even if you're not yet familiar with that call yet...

**Comment by ixti** [comment]
@rapind I can't recall at the moment but I remember I saw couple of absolutely different projects using same filename extensions for their files.

**Comment by ixti** [comment]
@minad Please merge this, if you have no objections.

**Comment by ixti** [comment]
This PR does not changes current behavior, but adds README description of it and how to change to be more "enforced".

**Comment by minad** [comment]
@ixti @genki @rapind I cannot really judge this and it would be good if we have someone maintaining the vim scripts. Is any one of you willing to do so?

**Comment by ixti** [comment]
@minad I'm not proficient in VimL to call myself a good candidate to be a maintainer, but I'm definitely will be happy to help. In any case, this particular change is not adding changes to the vim script itself ;)) it's adding explanation to the *README* why current version does not work in a very specific situation as one might expect, and how to avoid this.

**Comment by rapind** [comment]
@minad I've literally never even touched VimL. At most just tweaked my own configs. I don't even get a chance to use slim much these days either :(

**Comment by ixti** [comment]
Closing this due to no activity (I assume no interest as well) from maintainers side.

**Comment by minad** [comment]
@ixti As I wrote above, there is a need of someone maintaining this. Even if the maintainers are not very proficient in Viml, they could still take care that the PRs get tested and merged...

**Pull request closed without merging**